### PR TITLE
fix(gce): populate custom instance types in edit deploy stage

### PR DESCRIPTION
**Pull request by danielpeach**
@duftler or @jtk54 please review.

Closes https://github.com/spinnaker/spinnaker/issues/1516

@anotherchrisberry or @icfantv - this touches the core instance type service.

### CODE REVIEW

File: app/scripts/modules/core/instance/instanceType.service.ts [hunk 1]
```diff
@@ -85,7 +86,7 @@ export class InstanceTypeService {
 
   private getInstanceTypeCategory(cloudProvider: string, instanceType: string): ng.IPromise<IInstanceTypeCategory> {
     return this.getCategories(cloudProvider).then((categories: IInstanceTypeCategory[]) => {
-      return (categories || []).find(c => c.families.some(f => f.instanceTypes.some(i => i.name === instanceType)));
+      return (categories || []).find(c => c.families.some(f => f.instanceTypes.some(i => i.name === instanceType || (i.nameRegex && i.nameRegex.test(instanceType)))));
```
Comment (icfantv) [inline app/scripts/modules/core/instance/instanceType.service.ts]: I see what you're doing here, but this strikes me as funky.  We're adding a conditional that only looks at a property set by one cloud provider.

In OOP land, a class would override a method to provide the necessary implementation/business logic, but we're not _really_ in OOP land and we'd have to create a whole InstanceTypeServiceProvider mechanism to have the code serve up the "right" service...

I wonder if we could take advantage of the extracted settings by @jrsquared and customize this there instead?  Or is that totally overkill?

Comment (anotherchrisberry) [inline app/scripts/modules/core/instance/instanceType.service.ts]: I dunno, this makes sense to me. It doesn't affect other cloud providers, and provides a hook for future cloud providers to customize the name matching.

Comment (jrsquared) [inline app/scripts/modules/core/instance/instanceType.service.ts]: This is simple enough. I don't think we would buy anything by adding the complexity of using settings; it's already abstract.

Comment (icfantv) [inline app/scripts/modules/core/instance/instanceType.service.ts]: Yea, that's fair.

**Pull request merged**